### PR TITLE
fix: [PIE-5103]: remove coming soon banner and disabled unselected types on edit

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Removing the coming soon banner with CG deployment Types
- Removing the Currently available text above NG deployment Type
- Removing the disabled unselected deployment types from VISUAL view
- Handled the CSS issues with the Infrastructure Modal 

#### Screenshots

https://user-images.githubusercontent.com/106532291/187141886-b6e6887c-5d5f-42a4-9bdf-4a58465763af.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10829)

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 1]
```diff
@@ -78,26 +64,42 @@ const CardList = ({
...
-        return (
-          <Utils.WrapOptionalTooltip key={item.value} tooltipProps={item.tooltipProps} tooltip={item.tooltip}>
-            {allowDisabledItemClick ? (
-              <div onClick={() => onChange(item.value as ServiceDeploymentType)}>{itemContent}</div>
-            ) : (
-              itemContent
-            )}
-          </Utils.WrapOptionalTooltip>
-        )
+        let itemContent: JSX.Element = <></>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:67]: Both these conditions can be combined in one. 
`if (isReadonly && item.value === selectedValue || !isReadonly) ? () : ()`

Also you can spread the item value in map itself.

Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:67]: Added this in latest commit

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 2]
```diff
@@ -78,26 +60,31 @@ const CardList = ({
...
```
Comment (vkbansal-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:63]: use `filter` to filter out items

**Review by vkbansal-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 3]
```diff
@@ -78,26 +60,31 @@ const CardList = ({
...
+        {
+          ;((isReadonly && item.value === selectedValue) || !isReadonly) &&
+            (itemContent = (
+              <Thumbnail
+                key={item.value}
+                label={item.label}
+                value={item.value}
+                icon={item.icon}
+                disabled={item.disabled || isReadonly}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:72]: isReadyonly not needed here if you are not reaching here if isReadonly = true.
Also, I think you should show disabled deployment type tile if isReadonly = true

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 4]
```diff
@@ -78,26 +60,31 @@ const CardList = ({
...
+                selected={item.value === selectedValue}
+                onClick={handleChange}
+              />
+            ))
+        }
+        if (itemContent !== <></>)
+          return (
+            <Utils.WrapOptionalTooltip key={item.value} tooltipProps={item.tooltipProps} tooltip={item.tooltip}>
+              {allowDisabledItemClick ? (
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:81]: Please remove this. allowDisabledItemClick was true only for CG deployment type. For NG, we don't need to allow it.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx]: @vkbansal-harness addressed this in latest commit

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx]: Done as discussed

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx (hunk 4 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx]: @chetanvekariya-harness Addressed this in latest commit

**Review by vkbansal-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**